packages/app-mobile/components/NoteEditor/ImageEditor/ImageEditor.tsx

```diff
@@ -308,13 +318,16 @@ const ImageEditor = (props: Props) => {
 		const json = JSON.parse(data);
 		if (json.action === 'save') {
 			await clearAutosave();
-			props.onSave(json.data);
+			await props.onSave(json.data);
 		} else if (json.action === 'autosave') {
 			await writeAutosave(json.data);
 		} else if (json.action === 'save-toolbar') {
 			Setting.setValue('imageeditor.jsdrawToolbar', json.data);
 		} else if (json.action === 'close') {
 			onRequestCloseEditor(json.promptIfUnsaved);
+		} else if (json.action === 'save-and-close') {
+			await props.onSave(json.data);
+			onRequestCloseEditor(json.promptIfUnsaved);
```

Comment:
A new `save-and-close` action was added to fix an issue on iOS where clicking "Close" then "Save changes" would attach duplicates of a drawing. This saves before the editor starts loading (rather than during).